cli/medperf/comms/entity_resources/sources/synapse.py

```diff
@@ -0,0 +1,58 @@
+import synapseclient
+from synapseclient.core.exceptions import SynapseError
+from medperf.exceptions import (
+    CommunicationRetrievalError,
+    CommunicationAuthenticationError,
+)
+import os
+import shutil
+from .source import BaseSource
+import re
+
+
+class SynapseSource(BaseSource):
+    prefix = "synapse:"
+
+    @classmethod
+    def validate_resource(cls, value):
+        prefix = SynapseSource.prefix
+        if not value.startswith(prefix):
+            return
+
+        prefix_len = len(prefix)
+        value = value[prefix_len:]
+
+        if re.match(r"syn\d+$", value):
+            return value
+
+    def __init__(self):
+        self.client = synapseclient.Synapse()
+
+    def authenticate(self):
+        try:
+            self.client.login(silent=True)
+        except SynapseError as e:
+            msg = "There was an attempt to download resources from the Synapse "
```

Comment:
Is this actually the only possible type of error?

Reply:
All exceptions from the synapse client inherit from `SynapseError`.
I am following a convention that we only catch errors we know why they happen

Reply:
Sorry, I wasn't clear. I meant is this the only reason we would get a SynapseError, or could there be other reasons for a SynapseError?